OSDOCS-9882-cloud-compute: Bug batched the cloud compute bugs (#77709)
openshift/openshift-docs: release_notes/ocp-4-16-release-notes.adoc

[id="ocp-4-16-cloud-compute-bug-fixes_{context}"]
==== Cloud Compute

* Previously, a node associated with a rebooting machine briefly having a status of `Ready=Unknown` triggered the `UnavailableReplicas` condition in the Control Plane Machine Set Operator. This condition caused the Operator to enter the `Available=False` state and trigger alerts because that state indicates a nonfunctional component that requires immediate administrator intervention. This alert should not have been triggered for the brief and expected unavailabilty while rebooting. With this release, a grace period for node unreadiness is added to avoid triggering unnecessary alerts. (link:https://issues.redhat.com/browse/OCPBUGS-34970[*OCPBUGS-34970*])

> * Previously, a transient failure to fetch bootstrap data during machine creation, like a transient failure to connect to the API server, caused the machine to enter a terminal failed state. With this release, failure to fetch bootstrap data during machine creation is retried indefinitely until it eventually succeeds. (link:https://issues.redhat.com/browse/OCPBUGS-34158[*OCPBUGS-34158*])

Reviewer: ```suggestion
* Previously, a transient failure to fetch bootstrap data during machine creation, such as a transient failure to connect to the API server, caused the machine to enter a terminal failed state. With this release, failure to fetch bootstrap data during machine creation is retried indefinitely until it eventually succeeds. (link:https://issues.redhat.com/browse/OCPBUGS-34158[*OCPBUGS-34158*])
```

> * Previously, the `machine-api` operator panicked when deleting a server in an error state because it was not passed a port list. With this release, deleting a machine stuck in an `ERROR` state does not crash the controller anymore. (link:https://issues.redhat.com/browse/OCPBUGS-34155[*OCPBUGS-34155*])

Reviewer: ```suggestion
* Previously, the `machine-api` operator panicked when deleting a server in an error state because it was not passed a port list. With this release, deleting a machine stuck in an `ERROR` state does not crash the controller. (link:https://issues.redhat.com/browse/OCPBUGS-34155[*OCPBUGS-34155*])
```

* Previously, an optional internal function of the cluster autoscaler caused repeated log entries when it was not implemented. The issue is resolved in this release. (link:https://issues.redhat.com/browse/OCPBUGS-33932[*OCPBUGS-33932*])

> * Previously, if the control plane machine set was created with a template without a path during installation on a {vmw-first} cluster, the Control Plane Machine Set Operator rejected modifications or deletion of the control plane machine set custom resource (CR). With this release, the Operator allows template names for {vmw-short} in the control plane machine set definition. (link:https://issues.redhat.com/browse/OCPBUGS-32295[*OCPBUGS-32295*])

Reviewer: ```suggestion
* Previously, if the control plane machine set was created with a template without a path during installation on a {vmw-first} cluster, the Control Plane Machine Set Operator rejected modification or deletion of the control plane machine set custom resource (CR). With this release, the Operator allows template names for {vmw-short} in the control plane machine set definition. (link:https://issues.redhat.com/browse/OCPBUGS-32295[*OCPBUGS-32295*])
```

> * Previously, the Control Plane Machine Set Operator crashed when attempting to update a {vmw-first} cluster because the infrastructure resource was not configured. With this release, the Operator is able to handle this scenario so that the cluster update is able to proceed. (link:https://issues.redhat.com/browse/OCPBUGS-31808[*OCPBUGS-31808*])

Reviewer: ```suggestion
* Previously, the Control Plane Machine Set Operator crashed when attempting to update a {vmw-first} cluster because the infrastructure resource was not configured. With this release, the Operator is able to handle this scenario so that the cluster update can proceed. (link:https://issues.redhat.com/browse/OCPBUGS-31808[*OCPBUGS-31808*])
```

[id="ocp-4-16-cloud-compute-bug-fixes_{context}"]
==== Cloud Compute

* Previously, a node associated with a rebooting machine briefly having a status of `Ready=Unknown` triggered the `UnavailableReplicas` condition in the Control Plane Machine Set Operator. This condition caused the Operator to enter the `Available=False` state and trigger alerts because that state indicates a nonfunctional component that requires immediate administrator intervention. This alert should not have been triggered for the brief and expected unavailabilty while rebooting. With this release, a grace period for node unreadiness is added to avoid triggering unnecessary alerts. (link:https://issues.redhat.com/browse/OCPBUGS-34970[*OCPBUGS-34970*])

* Previously, a transient failure to fetch bootstrap data during machine creation, such as a transient failure to connect to the API server, caused the machine to enter a terminal failed state. With this release, failure to fetch bootstrap data during machine creation is retried indefinitely until it eventually succeeds. (link:https://issues.redhat.com/browse/OCPBUGS-34158[*OCPBUGS-34158*])

> * Previously, the `machine-api` operator panicked when deleting a server in an error state because it was not passed a port list. With this release, deleting a machine stuck in an `ERROR` state does not crash the controller. (link:https://issues.redhat.com/browse/OCPBUGS-34155[*OCPBUGS-34155*])

Reviewer 2: ```suggestion
* Previously, the Machine API Operator panicked when deleting a server in an error state because it was not passed a port list. With this release, deleting a machine stuck in an `ERROR` state does not crash the controller. (link:https://issues.redhat.com/browse/OCPBUGS-34155[*OCPBUGS-34155*])
```

Author: :heavy_check_mark:

* Previously, if the control plane machine set was created with a template without a path during installation on a {vmw-first} cluster, the Control Plane Machine Set Operator rejected modification or deletion of the control plane machine set custom resource (CR). With this release, the Operator allows template names for {vmw-short} in the control plane machine set definition. (link:https://issues.redhat.com/browse/OCPBUGS-32295[*OCPBUGS-32295*])

* Previously, the Control Plane Machine Set Operator crashed when attempting to update a {vmw-first} cluster because the infrastructure resource was not configured. With this release, the Operator can handle this scenario so that the cluster update is able to proceed. (link:https://issues.redhat.com/browse/OCPBUGS-31808[*OCPBUGS-31808*])

* Previously, when a user created a compute machine set with taints, they could choose to not specify the `Value` field. Failure to specify this field caused the cluster autoscaler to crash. With this release, the cluster autoscaler is updated to handle an empty `Value` field. (link:https://issues.redhat.com/browse/OCPBUGS-31421[*OCPBUGS-31421*])

> * Previously, IPv6 services were wrongly marked as internal on the OpenStack cloud provider, making it impossible to share IPv6 load balancers between {product-title} services. With this release, IPv6 services are not marked as internal, allowing IPv6 load balancers to be shared between services that use stateful IPv6 addresses. This fix allows load balancers to use stateful IPv6 addresses that are defined in the `loadBalancerIP` property of the service. (link:https://issues.redhat.com/browse/OCPBUGS-29605[*OCPBUGS-29605*])

Reviewer 2: ```suggestion
* Previously, IPv6 services were wrongly marked as internal on the {rh-openstack} cloud provider, making it impossible to share IPv6 load balancers between {product-title} services. With this release, IPv6 services are not marked as internal, allowing IPv6 load balancers to be shared between services that use stateful IPv6 addresses. This fix allows load balancers to use stateful IPv6 addresses that are defined in the `loadBalancerIP` property of the service. (link:https://issues.redhat.com/browse/OCPBUGS-29605[*OCPBUGS-29605*])
```

Author: :heavy_check_mark:
